packages/sacado/src/Sacado_Fad_GeneralFadTestingHelpers.hpp

```diff
@@ -0,0 +1,127 @@
+// @HEADER
+// ***********************************************************************
+//
+//                           Sacado Package
+//                 Copyright (2006) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// This library is free software; you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301
+// USA
+// Questions? Contact David M. Gay (dmgay@sandia.gov) or Eric T. Phipps
+// (etphipp@sandia.gov).
+//
+// ***********************************************************************
+//
+// @HEADER
+
+#ifndef SACADO_FAD_GENERALFADTESTINGHELPERS_HPP
+
+#ifndef TEUCHOS_TESTING_HELPERS_HPP
+#include "Teuchos_TestingHelpers.hpp"
+#endif
+
+#define SACADO_FAD_GENERALFADTESTINGHELPERS_HPP
```

Comment:
Why does this go after the inclusion of Teuchos_TestingHelpers.hpp?  Also all of this code should be #ifdef'd out if Sacado-TeuchosCore is not enabled by doing, e.g.,
```
#include "Sacado_ConfigDefs.h"
#ifdef HAVE_SACADO_TEUCHOSCORE
```